SdarotAPI/SdarotDriver.cs

```diff
@@ -58,48 +67,15 @@ public async Task<bool> Login(string username, string password)
 
     public async Task<IEnumerable<SeriesInformation>> SearchSeries(string searchQuery)
     {
-        var searchHtml = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.SearchUrl}{searchQuery}");
-        var doc = new HtmlDocument();
-        doc.LoadHtml(searchHtml);
-
-        var seriesNameElement = doc.DocumentNode.SelectSingleNode(Constants.XPathSelectors.SeriesPageSeriesName);
-
-        // In case there is only one result
-        if (seriesNameElement is not null)
-        {
-            var seriesName = seriesNameElement.InnerText.Trim(new[] { ' ', '/' });
+        var showsJson = await _httpClient.GetStringAsync($"{Constants.SdarotUrls.AllShowsUrl}");
+        var shows = JsonSerializer.Deserialize<List<SeriesInformation>>(showsJson);
 
-            var imageUrlElement = doc.DocumentNode.SelectSingleNode(Constants.XPathSelectors.SeriesPageSeriesImage);
-            if (imageUrlElement is null)
-            {
-                throw new ElementNotFoundException(nameof(imageUrlElement));
-            }
-
-            var imageUrl = imageUrlElement.GetAttributeValue("src", "");
-
-            return new SeriesInformation(HttpUtility.HtmlDecode(seriesName), imageUrl).Yield();
-        }
-
-        var seriesElements = doc.DocumentNode.SelectNodes(Constants.XPathSelectors.SearchPageResult);
-
-        // In case there are no results
-        if (seriesElements is null || seriesElements.Count == 0)
-        {
-            return Enumerable.Empty<SeriesInformation>();
-        }
+        var relevantShows = shows?.Where(x =>
+            x.SeriesNameHe.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase) ||
+            x.SeriesNameEn.Contains(searchQuery, StringComparison.CurrentCultureIgnoreCase)).ToList() ?? Enumerable.Empty<SeriesInformation>();
```

Comment:
The `relevantShows` type is `List`, so it shouldn't be assigned by an `Enumerable`. Move the `?? Enumerable...` to the return statement

Reply:
2 return statements under the same method are confusing. I think the base way forward is removing the .ToList(), keep everything IEnumerable and keep the code clean.
The `??` makes sure the relevantShows won't be nullable.
Tell me what you think.

Reply:
Sounds good. Just remove the `ToList()` then